src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -34,8 +34,12 @@ function forceUpdateIfMounted() {
 }
 
 function isControlled(props) {
-  var usesChecked = props.type === 'checkbox' || props.type === 'radio';
-  return usesChecked ? props.checked != null : props.value != null;
+  if (props.disabled) return false;
```

Comment:
The fact that a component is disabled does not mean it is uncontrolled.  This logic is not correct.